src/components/views/settings/FontScalingPanel.tsx

```diff
@@ -65,6 +65,14 @@ export default class FontScalingPanel extends React.Component<IProps, IState> {
         const client = MatrixClientPeg.safeGet();
         const userId = client.getSafeUserId();
         const profileInfo = await client.getProfileInfo(userId);
+        this.layoutWatcherRef = SettingsStore.watchSetting("layout", null, (...[, , , value]) => {
+            // Update the layout for the preview window according to the user selection
+            if (this.state.layout !== value) {
+                this.setState({
+                    layout: value,
```

Comment:
I don't think this is quite right either, as if the level at the account level changes this handler will fire, but if it is also set at the device level that will take precedence. I think here instead of using `value` you should just use `SettingsStore.getValue("layout")` again

Reply:
Although this works fine. 
```suggestion
        this.layoutWatcherRef = SettingsStore.watchSetting("layout", null, () => {
            // Update the layout for the preview window according to the user selection
            const value =  SettingsStore.getValue("layout")
            if (this.state.layout !== value) {
                this.setState({
                    layout: value,
```

But it has been implemented as this in other files.

```suggestion
       this.layoutWatcherRef = SettingsStore.watchSetting("layout", null, (...[, , , value]) =>
            this.setState({ layout: value as Layout }),
```